ksql-engine/src/main/java/io/confluent/ksql/engine/EngineExecutor.java

```diff
@@ -137,10 +138,15 @@ private void validateQuery(final QueryMetadata query, final ConfiguredStatement<
       final SchemaRegistryClient srClient = serviceContext.getSchemaRegistryClient();
 
       if (!AvroUtil.isValidSchemaEvolution(persistentQuery, srClient)) {
+        final org.apache.avro.Schema avroSchema = SchemaUtil.buildAvroSchema(
```

Comment:
`AvroUtil.isValidSchemaEvolution` internally builds the avro schema.  So it feels like duplication to also build it here.

This is the only place `isValidSchemaEvolution` is called.  I think a slight refactor of your change to move the exception below into the method would be better, as it avoids the need to build the schema twice. I'd rename the function to `void throwOnInvalidSchemaEvolution(queryMd, srClient)`.